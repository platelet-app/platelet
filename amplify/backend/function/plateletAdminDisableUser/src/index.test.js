const indexModule = require("./index");
const aws = require("aws-sdk");

const mockUser = {
    id: "userId",
    username: "username",
    tenantId: "tenantId",
    disabled: 0,
    _version: 1,
    _deleted: null,
    _lastChangedAt: 1620000000000,
};

jest.mock(
    "/opt/appSyncRequest",
    () => {
        return {
            request: jest.fn(),
            errorCheck: jest.fn(),
        };
    },
    { virtual: true }
);
jest.mock(
    "/opt/graphql/mutations",
    () => {
        return {
            updateUser: "updateUserMutation",
        };
    },
    { virtual: true }
);
jest.mock(
    "/opt/graphql/queries",
    () => {
        return {
            getUser: "getUserQuery",
        };
    },
    { virtual: true }
);

const appsyncModule = require("/opt/appSyncRequest");

// mock CognitoIdentityServiceProvider class
//
const cognitoIdentityServiceProvider = Object.getPrototypeOf(
    new aws.CognitoIdentityServiceProvider()
);
describe("plateletAdminDisableUser", () => {
    afterEach(() => jest.restoreAllMocks());
    test("disable a user", async () => {
        appsyncModule.request
            .mockResolvedValueOnce({
                json: () => ({
                    data: {
                        getUser: mockUser,
                    },
                }),
            })
            .mockResolvedValueOnce({
                json: () => ({
                    data: {
                        updateUser: { ...mockUser, disabled: 1 },
                    },
                }),
            });
        const cognitoDisableUserSpy = jest
            .spyOn(cognitoIdentityServiceProvider, "adminDisableUser")
            .mockReturnValue({
                promise: () => Promise.resolve(),
            });
        await indexModule.handler({ arguments: { userId: "userId" } });
        expect(appsyncModule.request).toMatchSnapshot();
        expect(cognitoDisableUserSpy).toMatchSnapshot();
    });
    test("disable a user appsync failure", async () => {
        appsyncModule.request
            .mockResolvedValueOnce({
                json: () => ({
                    data: {
                        getUser: mockUser,
                    },
                }),
            })
            .mockRejectedValueOnce(new Error("error"));
        const cognitoDisableUserSpy = jest
            .spyOn(cognitoIdentityServiceProvider, "adminDisableUser")
            .mockReturnValue({
                promise: () => Promise.resolve(),
            });
        const cognitoEnableUserSpy = jest
            .spyOn(cognitoIdentityServiceProvider, "adminEnableUser")
            .mockReturnValue({
                promise: () => Promise.resolve(),
            });
        await expect(
            indexModule.handler({ arguments: { userId: "userId" } })
        ).rejects.toThrow();
        expect(appsyncModule.request).toMatchSnapshot();
        expect(cognitoDisableUserSpy).toMatchSnapshot();
        expect(cognitoEnableUserSpy).toMatchSnapshot();
    });
});
