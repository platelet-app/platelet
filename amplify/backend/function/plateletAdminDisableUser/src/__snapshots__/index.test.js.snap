// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plateletAdminDisableUser disable a user 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "query": "getUserQuery",
        "variables": {
          "id": "userId",
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "updateUserMutation",
        "variables": {
          "input": {
            "_version": 1,
            "disabled": 1,
            "id": "userId",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`plateletAdminDisableUser disable a user 2`] = `
[MockFunction] {
  "calls": [
    [
      {
        "UserPoolId": "somePoolId",
        "Username": "username",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "promise": [Function],
      },
    },
  ],
}
`;

exports[`plateletAdminDisableUser disable a user appsync failure 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "query": "getUserQuery",
        "variables": {
          "id": "userId",
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "updateUserMutation",
        "variables": {
          "input": {
            "_version": 1,
            "disabled": 1,
            "id": "userId",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`plateletAdminDisableUser disable a user appsync failure 2`] = `
[MockFunction] {
  "calls": [
    [
      {
        "UserPoolId": "somePoolId",
        "Username": "username",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "promise": [Function],
      },
    },
  ],
}
`;

exports[`plateletAdminDisableUser disable a user appsync failure 3`] = `
[MockFunction] {
  "calls": [
    [
      {
        "UserPoolId": "somePoolId",
        "Username": "username",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "promise": [Function],
      },
    },
  ],
}
`;
